src/gval/homogenize/rasterize.py

```diff
@@ -0,0 +1,104 @@
+"""
+Functions to rasterize vector datasets to process in raster space
+"""
+
+from typing import Union
+
+import numpy as np
+import xarray as xr
+import geopandas as gpd
+from geocube.api.core import make_geocube
+
+
+def _rasterize_data(
+    candidate_map: Union[xr.Dataset, xr.DataArray],
+    benchmark_map: Union[gpd.GeoDataFrame, xr.Dataset, xr.DataArray],
+    rasterize_attributes: list,
+) -> Union[xr.Dataset, xr.DataArray]:
+    """
+
+    Parameters
+    ----------
+    candidate_map: Union[xr.Dataset, xr.DataArray]
+        Candidate map to reference in creation of rasterized benchmark map
+    benchmark_map: Union[gpd.GeoDataFrame, xr.Dataset, xr.DataArray]
+        Benchmark map to be rasterized
+    rasterize_attributes: list
+        Attributes to rasterize
+
+    Returns
+    -------
+    Union[xr.Dataset, xr.DataArray]
+        Rasterized Data
+
+    Raises
+    ------
+    KeyError
+
+    """
+
+    if isinstance(benchmark_map, gpd.GeoDataFrame):
+        rasterized_data = make_geocube(
+            vector_data=benchmark_map,
+            measurements=rasterize_attributes,
+            like=candidate_map,
+        )
+
+        if len(rasterized_data.data_vars) < 1:
```

Comment:
This checks the variables to ensure they are numeric? Can we do this prior to geocubing?

Reply:
Checks are now prior to geocube